app.js

```diff
@@ -1,31 +1,74 @@
 import { getUser, signOut } from './services/auth-service.js';
 import { protectPage } from './utils.js';
 import createUser from './components/User.js';
+import createShoppingForm from './components/ShoppingForm.js';
+import { addItem, allItems, updateItem, deleteItem } from './services/list-service.js';
+import createItems from './components/Items.js';
 
 // State
 let user = null;
+let items = [];
 
 // Action Handlers
 async function handlePageLoad() {
     user = getUser();
     protectPage(user);
 
+    items = await allItems();
+    // console.log(data); 
     display();
 }
 
 async function handleSignOut() {
     signOut();
 }
 
+async function handleAdd(item, quantity) {
+    // console.log(item, quantity);
+    const itemsList = { item, quantity, user_id: user.id };
+    const data = await addItem(itemsList);
+
+
+    items.push(data);
+
+    display();
+}
+
+async function handleUpdate(item) {
+    item = !item;
+    const index = items.indexOf(item);
+    item[index] = await updateItem(item);
```

Comment:
On this line, you're trying to save the supabase response into an array called item at a particular index. You don't need to do that, since you call your display function on the next line and this array will get overwritten with the info from the database. Go ahead and remove everything before the 'await' keyword.